src/sage/matrix/matrix_polynomial_dense.pyx

```diff
@@ -4079,3 +4084,462 @@ cdef class Matrix_polynomial_dense(Matrix_generic_dense):
                 if P[j, j].degree() + shifts[j] <= degree_bound:
                     column_indices.append(j)
             return P[:,column_indices]
+
+    def _basis_completion_via_reversed_approx(self):
+        r"""
+        Return a Smith form-preserving nonsingular completion of a row basis of
+        this matrix. For a more detailed description, see
+        :meth:`basis_completion`, in the row-wise case.
+
+        EXAMPLES::
+
+            sage: pR.<x> = GF(7)[]
+        """
+        from sage.matrix.constructor import matrix # for identity
+
+        ring,var = self.base_ring().objgen()
+        m = self.nrows()
+        n = self.ncols()
+
+        # corner cases: after this, m>0 and n>0
+        if m == 0 :
+            return matrix.identity(ring, n)
+        if n == 0 :
+            return matrix(ring, 0, 0)
+
+        # find column degrees (zero columns have degree -1)
+        cdeg = self.column_degrees()
+
+        # list zero and nonzero columns
+        zcols = []
+        nonzcols = []
+        for j in range(n):
+            if cdeg[j] < 0:
+                zcols.append(j)
+            else :
+                nonzcols.append(j)
+
+        if len(nonzcols) == 0 :
+            return matrix.identity(ring, n).matrix_from_rows(zcols)
```

Comment:
```suggestion
        if len(nonzcols) == 0 :
            return matrix.identity(ring, n)
```

If `nonzcols` is empty, then `zcols == list(range(n))`, so the `.matrix_from_rows` part is nop.

Reply:
Absolutely correct, thank you!